File: backend/app/domain/models/dose_record.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.infrastructure.database.connection import Base

class DoseRecord(Base):
    __tablename__ = "registros_dose"

    id                      = Column(Integer, primary_key=True, autoincrement=True)
    paciente_medicamento_id = Column(Integer, ForeignKey("paciente_medicamentos.id"), nullable=False)
    horario_medicacao_id    = Column(Integer, ForeignKey("horarios_medicacao.id"))
    registrado_por          = Column(Integer, ForeignKey("usuarios.id"))
    data_hora_prevista      = Column(DateTime, nullable=False)
    data_hora_confirmada    = Column(DateTime)
    status                  = Column(String, nullable=False)  # TOMADO | PENDENTE | PULADO | ATRASADO
    observacao              = Column(Text)
    metodo_confirmacao      = Column(String)  # MANUAL | QR_CODE | NFC

    # Relationships
    paciente_medicamento = relationship("PatientMedication", foreign_keys=[paciente_medicamento_id])
    horario_medicacao = relationship("MedicationSchedule", foreign_keys=[horario_medicacao_id])
    registrado_por_rel = relationship("User", foreign_keys=[registrado_por])

File: frontend/capsule-mobile/src/types/doseRecord.ts
export interface DoseRecordCreate {
  paciente_medicamento_id: number;
  status: string; 
  data_hora_prevista: string;
  observacao?: string | null;
  horario_medicacao_id?: number | null;
  registrado_por?: number | null;
  metodo_confirmacao?: string | null;
}

export interface DoseRecordResponse extends DoseRecordCreate {
  id: number;
  data_hora_confirmada?: string | null;
}